pydra/engine/helpers.py

```diff
@@ -544,3 +515,96 @@ def get_available_cpus():
 
     # Last resort
     return os.cpu_count()
+
+
+def load_and_run(
+    task_pkl, ind=None, rerun=False, submitter=None, plugin=None, **kwargs
+):
+    """
+     loading a task from a pickle file, settings proper input
+     and running the task
+     """
+    task = load_task(task_pkl=task_pkl, ind=ind)
+    try:
+        task(rerun=rerun, plugin=plugin, submitter=submitter, **kwargs)
+    except Exception as excinfo:
+        etype, eval, etr = sys.exc_info()
+        traceback = format_exception(etype, eval, etr)
+        crashfile = report_crash(task, traceback=traceback)
+        raise type(excinfo)(
+            str(excinfo.with_traceback(None)),
+            f" full crash report is here: {crashfile}",
+        )
```

Comment:
there could be 3 scenarios that could error:
1. load_task
2. preparation before a task runs in the __call__ function of the task
3. the task itself.

a part of 3 saves records an error (see the `_run` function in TaskBase. it also creates a result file with `errored=True`

this except should handle the other scenarios and not overwrite the result file. it should create one with `errored=True` if one doesn't exist and change the existing one if necessary.

Reply:
how to you want to get to point 2??

Reply:
this error will be raised before even task is saved in a pickle file

Reply:
it's part of `__call__` of TaskBase and all the way till the try-except clause in `_run`. many things can fail along the way till that try in `_run`

Reply:
>this error will be raised before even task is saved in a pickle file
i understand crashfile is generated later, but this is not about generating the crashfile, it's about trapping all possible errors in the code prior.

Reply:
so if `_result.pklz` exists you don't want me to overwrite? and what should be there if it doesn't exist? do you wanta traceback, or just `Result(output=None, runtime=None, errored=True)`?

Reply:
> so if _result.pklz exists you don't want me to overwrite?
correct (it also means the exception was raised by `_run `

> what should be there if it doesn't exist? do you wanta traceback, 
yes, you can record_error in the same way run does

> or just Result(output=None, runtime=None, errored=True)?
yes, return just an error.